content/docs/user-guide/programming/messaging/overview.md

```diff
@@ -0,0 +1,24 @@
+- `AZ::Interface`: A C++ template class for a global interface that other components can invoke requests from. 
+- `AZ::Event`: A C++ template class that can publish single value messages, which other components can subscribe to. 
+
+These three modules overlap some of their functionality: `AZ::Interface` provides a request interface, `AZ::Event` provides a publish-subscribe interface, and EBus provides both. However, each has strengths and weaknesses that make them more suitable for different scenarios. 
+
+At a glance, the following table compares some key characteristics of EBus, `AZ::Interface`, and `AZ::Event`. All three event messaging modules can be found throughout O3DE's code base.
+
+| | `AZ::Interface` | `AZ::Event` | EBus |
+| --- | --- | --- | --- |
+| Communication model | An `AZ::Interface<T>` template specialization defines a global singleton-like request interface, which other systems can invoke. Other systems can simply access the request interface by calling `AZ::Interface<T>.Get()`.  | An `AZ::Event` instance must be attached to a source component. Therefore, other components must have a reference to the source component to subscribe to the event. Other components subscribe to an event by implementing a handler that will process the event when it's signaled. An event can be signaled by calling `Signal()`, which subscribers can listen to and process via their event handler. A handler can only connect to one event, but an event can have multiple handlers. Events are useful for components talking to other components within the same entity. For cross-entity communication, you can implement `AZ::Event`s on an `AZ::Interface`, or use an EBus. | EBus is a general purpose event messaging system whose design focuses on abstraction and decoupling systems. It allows many source components to provide requests or publish messages, and many listener components to invoke those requests or subscribe to those messages - all without the components having to know about each other. Unlike `AZ::Event`, a listener can subscribe to an EBus specialization even if it doesn't exist yet. |
+| Lifetime | There can only be a single instance of an `AZ::Interface<T>` specialization at a time. It initializes when the application launches and gets destroyed when the application is terminated. | There can be many instances of an `AZ::Event<T>` specialization. Each instance must be attached to a component. It initializes when the component is created and gets destroyed when the component is destroyed. | EBus is a global singleton. It initializes when the application launches and gets destroyed when the application terminates. |
+| Scripting |  |  | Unlike `AZ::Interface`, only EBus can connect to scripting. This allows your component to expose its functionality to O3DE's scripting modules, such as Script Canvas and Lua. |
```

Comment:
There exists script-bind support for AZ::Event. It may be good to include that information here.